Add -cudaXXX to dali package name (#1948)
NVIDIA/DALI: docs/installation.rst

Installation
^^^^^^^^^^^^
To install a CUDA 9.0 based DALI build, execute the following command:

.. code-block:: bash

    pip install --extra-index-url https://developer.download.nvidia.com/compute/redist nvidia-dali-cuda90

For CUDA 10.0 based build use:

.. code-block:: bash

   pip install --extra-index-url https://developer.download.nvidia.com/compute/redist nvidia-dali-cuda100

.. note::

>   Since version 0.6.1, DALI doesn't contain prebuilt versions of the DALI TensorFlow plugin. It needs to be installed as a separate package which will be built against the currently installed version of TensorFlow:

Author: I was thinking to do that. @JanuszL what do you think?

>   Since version 0.6.1, DALI doesn't contain prebuilt versions of the DALI TensorFlow plugin. It needs to be installed as a separate package which will be built against the currently installed version of TensorFlow:

Reviewer: Why now. Go ahead.

.. code-block:: bash

   pip install --extra-index-url https://developer.download.nvidia.com/compute/redist nvidia-dali-tf-plugin-cuda90

.. code-block:: bash

   pip install --extra-index-url https://developer.download.nvidia.com/compute/redist nvidia-dali-tf-plugin-cuda100

> Installing this package will install ``nvidia-dali-cudaXXX`` and its dependencies, if these dependencies are not already installed. The package ``tensorflow-gpu`` must be installed before attempting to install ``nvidia-dali-tf-plugin-cudaXXX``.

Reviewer 2: ```suggestion
Installing this package will install ``nvidia-dali-cudaXXX`` and its dependencies, if they are not already installed. The package ``tensorflow-gpu`` must be installed before attempting to install ``nvidia-dali-tf-plugin-cudaXXX``.
```

Author: Will do

.. note::

>   The package ``nvidia-dali-tf-plugin-cudaXXX`` has a strict requirement with ``nvidia-dali-cudaXXX`` as its exact same version.
>   Thus, installing ``nvidia-dali-tf-plugin-cudaXXX`` at its latest version will replace any older ``nvidia-dali-cudaXXX`` versions already installed with the latest.

Reviewer 2: Can we paraphrase these two a little bit? Reading those makes my mind dry xD
```suggestion
  The packages ``nvidia-dali-tf-plugin-cudaXXX`` and ``nvidia-dali-cudaXXX`` shall be in exactly the same version.
  Therefore, installing the latest ``nvidia-dali-tf-plugin-cudaXXX``, will replace any older ``nvidia-dali-cudaXXX`` version already installed.
```

Author: Will do

  To work with older versions of DALI, provide the version explicitly to the ``pip install`` command.

.. code-block:: bash

    CUDA_SUFFIX=100

>     CURRENT_DALI_VERSION=0.23  # Already installed version of nvidia-dali-cuda${CUDA_SUFFIX}, assuming we don't want to upgrade

Reviewer 2: That's a nitpick, feel free to ignore ;)
```suggestion
    CURRENT_DALI_VERSION=0.23  # Already installed version of nvidia-dali-cuda${CUDA_SUFFIX}, assuming you don't want to upgrade it
```

Author: Will do

    pip install --extra-index-url https://developer.download.nvidia.com/compute/redist nvidia-dali-tf-plugin-cuda${CUDA_SUFFIX}==$CURRENT_DALI_VERSION

> For older versions of DALI (0.22 and lower), use the package `nvidia-dali`. The CUDA version is selected by changing the pip index:

Reviewer 2: ```suggestion
For older versions of DALI (0.22 and lower), use the package `nvidia-dali`. The CUDA version can selected by changing the pip index:
```

Author: "can be" you mean, I guess. Will do

  Since version 0.6.1, DALI doesn't contain prebuilt versions of the DALI TensorFlow plugin. It needs to be installed as a separate package which will be built against the currently installed version of TensorFlow:

* for CUDA 9:

.. code-block:: bash

   pip install --extra-index-url https://developer.download.nvidia.com/compute/redist nvidia-dali-tf-plugin-cuda90

* for CUDA 10:

.. code-block:: bash

   pip install --extra-index-url https://developer.download.nvidia.com/compute/redist nvidia-dali-tf-plugin-cuda100

Installing this package will install ``nvidia-dali-cudaXXX`` and its dependencies, if they are not already installed. The package ``tensorflow-gpu`` must be installed before attempting to install ``nvidia-dali-tf-plugin-cudaXXX``.

.. note::

  The packages ``nvidia-dali-tf-plugin-cudaXXX`` and ``nvidia-dali-cudaXXX`` should be in exactly the same version.
  Therefore, installing the latest ``nvidia-dali-tf-plugin-cudaXXX``, will replace any older ``nvidia-dali-cudaXXX`` version already installed.
  To work with older versions of DALI, provide the version explicitly to the ``pip install`` command.

.. code-block:: bash

    CUDA_SUFFIX=100

>     CURRENT_DALI_VERSION=0.23  # Already installed version of nvidia-dali-cuda${CUDA_SUFFIX}, assuming you don't want to upgrade it

Reviewer 3: Why we use current DALI version here? I don't get this comment.

Author: I just want to show how to install nvidia-dali-tf-plugin for a particular DALI version that is already installed. That is, without upgrading to latest. I can't put 0.6.1 there or something like that because the package name won't exist till 0.23, that's why.

I am not sure if we should keep this example here. I don't know  how useful it is. What do you think?

Author: I removed this snippet
